components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/dao/AuthorizationCodeDAOImplTest.java

```diff
@@ -0,0 +1,463 @@
+import static org.mockito.Matchers.anyObject;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.doNothing;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.wso2.carbon.identity.oauth2.util.OAuth2Util.getTenantId;
+
+/**
+ * Unit tests for AuthorizationCodeDAOImpl.
+ */
+@WithCarbonHome
+@PrepareForTest({IdentityDatabaseUtil.class, OAuth2Util.class, OAuth2TokenUtil.class, IdentityUtil.class,
+        OAuthServerConfiguration.class})
+public class AuthorizationCodeDAOImplTest extends PowerMockIdentityBaseTest {
+    public static Map<String, BasicDataSource> dataSourceMap = new HashMap<>();
+
+    @Mock
+    private ServiceProvider mockedServiceProvider;
+
+    @Mock
+    private AuthenticatedUser mockedAuthenticatedUser;
+
+    @Mock
+    private ApplicationManagementService applicationManagementService;
+
+    private AuthorizationCodeDAOImpl authorizationCodeDAO;
+    private String[] scopes;
+    AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+    private static final int DEFAULT_TENANT_ID = 1234;
+    private static final String APP_NAME = "myApp";
+    private static final String USER_NAME = "user1";
+    private static final String CALLBACK = "http://localhost:8080/redirect";
+    private static final String DB_NAME = "testAuthzCODEDB";
+
+    @BeforeClass
+    public void initTest() throws Exception {
+
+        //Initializing database
+        DAOUtils.initializeDataSource(DB_NAME, DAOUtils.getFilePath("identity.sql"));
+        authorizationCodeDAO = new AuthorizationCodeDAOImpl();
+        scopes = new String[]{"sms", "openid", "email"};
+        authenticatedUser.setTenantDomain("super.wso2");
+        authenticatedUser.setUserName("randomUser");
+        authenticatedUser.setUserStoreDomain("PRIMARY");
+        storeIDP();
+    }
+
+    @AfterClass
+    public void tearDown() throws Exception {
+        closeH2Base(DB_NAME);
+    }
+
+    public static void closeH2Base(String databaseName) throws Exception {
+        BasicDataSource dataSource = dataSourceMap.get(databaseName);
+        if (dataSource != null) {
+            dataSource.close();
+        }
+    }
+
+    private AuthzCodeDO persistAuthorizationCode(String consumerKey, String authzCodeId, String authzCode,
+                                                 boolean createApplication, String status) throws Exception {
+        if (createApplication) {
+            createApplication(consumerKey, UUID.randomUUID().toString(), DEFAULT_TENANT_ID);
+        }
+        AuthzCodeDO authzCodeDO = new AuthzCodeDO(authenticatedUser, scopes, new Timestamp(System.currentTimeMillis()),
+                3600000L, CALLBACK, consumerKey, authzCode, authzCodeId, status, null,
+                null);
+        try (Connection connection = DAOUtils.getConnection(DB_NAME)) {
+            prepareConnection(connection);
```

Comment:
We have closed the DB connection in the tearDown() method. Similarly, can't we create the DB connection in the setup method?

Reply:
Addressed